### PR TITLE
Add frontend smoke tests for annotations

### PR DESCRIPTION
**Pull request by mweiden**
Partially completes https://github.com/chanzuckerberg/cellxgene/issues/969

This PR covers the following test cases (basically all the frontend tests):
- make sure the system works with anno turned off (experimental labels flag off) — package.json smoke-test-without-annotation
- with flags on, do the normal tests pass — package.json smoke-test-with-annotation
- create a category
- create a label
-  assign cells to labels and make sure numbers are right
- delete a label
- delete categories
- rename a label
- rename a category
- if a label has cells, select a different set and assign it
- subset the world, all the same tests, but where universe is not equal to world

It does not cover the following test cases from https://github.com/chanzuckerberg/cellxgene/issues/969:
- make sure that the stuff saved in the file is what you expect, was everything saved (pytest)
- checks that you load the file correctly on startup (pytest)
- all the above tests, but throw in some undo redo, (add a category, undo, then redo)

Note the undo/redo test is blocked by https://github.com/chanzuckerberg/cellxgene/issues/1153

### CODE REVIEW

**Comment by mweiden** [comment]
@bkmartinjr when you review this, please note f0bd970. Motivation: when running the test the annotations fixture can update, which causes the test to brake.

**Comment by bkmartinjr** [comment]
> @bkmartinjr when you review this, please note [f0bd970](https://github.com/chanzuckerberg/cellxgene/commit/f0bd970bb2b9b7a3ec6a5c15bebcab9e7f2684a5). Motivation: when running the test, autosave can update the annotations fixture, which causes the subsequent runs of the test to break.


@mweiden - what do you mean by "fixture"?   Also, down the road, if saving is completely disabled, how will you do a full test that the annotations make it into the server-side correctly (ie, are saved into the CSV)?

**Comment by mweiden** [comment]
> what do you mean by "fixture"?

@bkmartinjr I'd call `example-dataset/pbmc3k-annotations.csv` and `example-dataset/pbmc3k.h5ad` "test fixtures" since they are both starting datasets used in testing.

> if saving is completely disabled, how will you do a full test that the annotations make it into the server-side correctly (ie, are saved into the CSV)?

This is a good question and I share the concern. Something here needs to bend: even if we construct the test categories/labels before each test, reproducibility will be a problem since the server will periodically save state.

One option: have the script that runs the test reset any altered test fixtures or remove any saved annotations state (in the case of constructing them on the fly).

**Comment by bkmartinjr** [comment]
> One option: have the script that runs the test reset (git checkout ...) any altered test fixtures or remove any saved annotations state (in the case of constructing them on the fly).

Another option: have the test make a copy of the base file to /tmp, and start the server pointing at the temp file.  Probably need to generate a unique tmpfile name in case the test gets run more than once, concurrently.

**Comment by mweiden** [comment]
@bkmartinjr I like it. I'll give that a shot.

File: client/src/globals.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { Colors } from "@blueprintjs/core";
 import { dispatchNetworkErrorMessageToUser } from "./util/actionHelpers";
 
+/* flags for testing */
+export const autosaveEnabled = process.env.AUTOSAVE_ENABLED ? process.env.AUTOSAVE_ENABLED === "true" : true;
```
Comment (bkmartinjr) [inline client/src/globals.js]: One last-minute thought:  I think we will need to remove this at some point, as I can't see a world where we (literally) don't save a user's annotations.

Can you give it a name more suggestive of its temporary "work around" nature?

Eg, "tmpWorkAroundAutosaveEnabled"?

Comment (mweiden) [reply client/src/globals.js]: I'm just going to implement the `/tmp` file idea and push it to this PR here in a bit.

**Pull request merged**